Sources/App/Controllers/PackageController.swift

```diff
@@ -17,4 +17,15 @@ struct PackageController {
             .map { PackageShow.View($0).document() }
     }
 
+    func _show(req: Request) throws -> EventLoopFuture<HTML> {
+        guard
+            let owner = req.parameters.get("owner"),
+            let repository = req.parameters.get("repository")
+            else {
+                return req.eventLoop.future(error: Abort(.notFound))
+        }
+        return PackageShow.Model.query(database: req.db, owner: owner, repository: repository)
+            .map { PackageShow.View($0).document() }
+    }
```

Comment:
In this PR I'm leaving the `/packages/{id}` route in and add the new `_show` and `_packages(_ owner:_ repository)` route with `_` for now.

I've had a quick look at taking the id based lookup out but this will require a set of further changes:

- RecentPackages:
  - update mat. view and model
  - update link generation on homepage
- RecentReleases
  - update mat. view and model
  - update link generation on homepage
- Search links (JS)

This would make for quite a big PR.

It's also a risky PR, because we don't have coverage for all the required the link changes.

Reply:
That's fine for this PR, but let's consider this change fairly high priority feature as that'll mean those URLs have been public if we preview/launch with that.

Search links JS should be done via #177 which will also centralise the generation of URLs